python/src/lib.rs

```diff
@@ -318,6 +319,35 @@ impl RawDeltaTable {
         Ok(serde_json::to_string(&metrics).unwrap())
     }
 
+    // Run the restore command on the Delta Table: restore table to a given version or datetime
+    #[pyo3(signature = (version_to_restore = None, datetime_to_restore = None, ignore_missing_files = false, protocol_downgrade_allowed = false))]
+    pub fn restore(
+        &mut self,
+        version_to_restore: Option<i64>,
+        datetime_to_restore: Option<&str>,
+        ignore_missing_files: bool,
+        protocol_downgrade_allowed: bool,
+    ) -> PyResult<String> {
+        let mut cmd = RestoreBuilder::new(self._table.object_store(), self._table.state.clone());
+        if let Some(version) = version_to_restore {
+            cmd = cmd.with_version_to_restore(version)
+        }
+        if let Some(ds) = datetime_to_restore {
```

Comment:
Since both of these parameters are optional there is a potential for a user to inadvertently provide _both_ arguments and that would result in two restore "commands" being passed into the builder, and the `datetime` one is the one which will win in that case.

I think it might be useful to raise an error here to indicate to the user that they should not provide both parameters, or issue a warning that the `version_to_restore` will be ignored.

Reply:
This check has been done here.[parameter check](https://github.com/delta-io/delta-rs/blob/main/rust/src/operations/restore.rs#L132). Should I do double check?

Reply:
I think the check is good. But TBH, for the Python interface, I think we should combine the parameters into one. See my other comment.